crypto/bn256/gnark/g1.go

```diff
@@ -0,0 +1,51 @@
+package bn256
+
+import (
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+)
+
+// G1 is the affine representation of a G1 group element.
+//
+// Since this code is used for precompiles, using Jacobian
+// points are not beneficial because there are no intermediate
+// points to allow us to save on inversions.
+//
+// Note: We also use this struct so that we can conform to the existing API
+// that the precompiles want.
+type G1 struct {
+	inner bn254.G1Affine
```

Comment:
You can omit `inner` (keeping `bn254.G1Affine`).  Then access the internal instance by `instanceName.G1Affine.instanceMethodOrField`.  I think that would be the proper Go way to do this.

Reply:
But that would also mean we are exposing the public api to other packages. 
I think the idea was to wrap the existing api and make it a drop in replacement for google and cloudflare

Reply:
Oh.  Yeah true.  We should minimize the API surface.